documentation/DEVELOPER-GUIDELINES.md

```diff
@@ -35,7 +35,11 @@ In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines
   - The `ConfigError` enum should be separate from other `Error` enums used by the driver.
   - The driver should implement `fn apply_config(&mut self, config: &Config) -> Result<(), ConfigError>`.
   - In case the driver's configuration is infallible (all possible combinations of options are supported by the hardware), the `ConfigError` should be implemented as an empty `enum`.
-  - Configuration structs should derive `procmacros::BuilderLite` in order to automatically implement the Builder Lite pattern for them.
+  - Configuration structs should
+    - Use the [Builder Lite](https://matklad.github.io/2022/05/29/builder-lite.html) pattern. Prefer deriving `procmacros::BuilderLite` in order to automatically implement the pattern.
+    - Implement `Copy` if possible.
+    - The fields should be private. The `BuilderLite` macro generates setters and getters automatically.
```

Comment:
Why mandate this for every config struct? public fields are so much nicer to work with compare to getters and setters.
If the struct is simple enough couldn't the just remain public?

Reply:
What about cases where the struct config is complete? 
Does consistency still trump that?

Reply:
Ah, is this rule recursive too? So structs in the config struct also have to have setters and getters, etc.

Reply:
The problem with consistency is that you have to be consistent :D. So yes, I understand that for some structs this will be an artificial limitation, but I'm pretty sure most structs _won't_ be complete and special casing them feels odd to me. _maybe_ I can see a world where we make those structs have public fields but we would always have the getters/setters as a consistency fall back. We can do this at anytime post 1.0 of course.

Reply:
> Ah, is this rule recursive too? So structs in the config struct also have to have setters and getters, etc.

Good question. I don't think we need to go that far, but if a sub struct needs private fields for whatever reason then they have to go this way anyways. Did you have a config structure in mind that might have this property?

Reply:
> Did you have a config structure in mind that might have this property?

The one for the DPI driver